tarhelper/untar.go

```
@@ -258,11 +260,23 @@ func (u *Untar) processEntry(header *tar.Header) error {
 
 	// resolve the destination and then reset the name based on the resolution
 	destDir, err := u.resolveDestination(filepath.Dir(name))
-	name = filepath.Join(destDir, filepath.Base(name))
 	if err != nil {
 		return err
 	}
 
+	name = filepath.Join(destDir, filepath.Base(name))
+
+	// The path length of the extracted file might exceed Windows maximum of
+	// 260 chars.
+	if runtime.GOOS == "windows" {
+		absPath, err := filepath.Abs(name)
+		// If we can't get the absolute path, just assume it's less than 260
+		// chars. The assumption will be correct most of the time.
```

Yeah, I'd rather return the error instead of assuming. Why would the `Abs` function fail?